internal/test/resources.go

```diff
@@ -319,6 +319,115 @@ func NewCryostatWithReportsResources() *operatorv1beta1.Cryostat {
+		NodeSelector: map[string]string{"node": "good"},
+		Affinity: &operatorv1beta1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+					NodeSelectorTerms: []corev1.NodeSelectorTerm{
+						{
+							MatchExpressions: []corev1.NodeSelectorRequirement{
+								{
+									Key:      "node",
+									Operator: corev1.NodeSelectorOpIn,
+									Values: []string{
+										"good",
+										"better",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			PodAffinity: &corev1.PodAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+					{
+						LabelSelector: metav1.AddLabelToSelector(&metav1.LabelSelector{},
+							"pod", "good"),
+						TopologyKey: "topology.kubernetes.io/zone",
+					},
+				},
+			},
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+					{LabelSelector: metav1.AddLabelToSelector(&metav1.LabelSelector{},
+						"pod", "bad"),
+						TopologyKey: "topology.kubernetes.io/zone",
+					},
+				},
+			},
+		},
+		Tolerations: []corev1.Toleration{
+			{
+				Key:      "node",
+				Operator: corev1.TolerationOpEqual,
+				Value:    "ok",
+				Effect:   corev1.TaintEffectNoExecute,
+			},
+		},
+	}
+	return cr
+}
+
+func NewCryostatWithReportsScheduling() *operatorv1beta1.Cryostat {
+	cr := NewCryostat()
+	cr.Spec.ReportOptions = &operatorv1beta1.ReportConfiguration{
+		SchedulingOptions: &operatorv1beta1.SchedulingConfiguration{
+			NodeSelector: map[string]string{"node": "good"},
+			Affinity: &operatorv1beta1.Affinity{
+				NodeAffinity: &corev1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+						NodeSelectorTerms: []corev1.NodeSelectorTerm{
+							{
+								MatchExpressions: []corev1.NodeSelectorRequirement{
+									{
+										Key:      "node",
+										Operator: corev1.NodeSelectorOpIn,
+										Values: []string{
+											"good",
+											"better",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				PodAffinity: &corev1.PodAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{
+							LabelSelector: metav1.AddLabelToSelector(&metav1.LabelSelector{},
+								"pod", "good"),
+							TopologyKey: "topology.kubernetes.io/zone",
+						},
+					},
+				},
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{LabelSelector: metav1.AddLabelToSelector(&metav1.LabelSelector{},
+							"pod", "bad"),
+							TopologyKey: "topology.kubernetes.io/zone",
+						},
+					},
+				},
+			},
+			Tolerations: []corev1.Toleration{
+				{
+					Key:      "node",
+					Operator: corev1.TolerationOpEqual,
+					Value:    "ok",
+					Effect:   corev1.TaintEffectNoExecute,
+				},
+			},
+		},
```

Comment:
Since these structs are the same, I would extract the SchedulingConfiguration to a helper function and call that to populate `spec.SchedulingOptions` and `spec.ReportOptions.SchedulingOptions` in `NewCryostatWithScheduling` and `NewCryostatWithReportsScheduling` respectively.